api/schema/application_aws_resources_resource.go

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Illumio, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package schema
+
+import (
+	resource_schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	applicationAwsResourcesResource = Resource{
+		TypeName: "application_aws_resources",
+		Schema: resource_schema.Schema{
+			Version:     1,
+			Description: "Resources associated with an application in the Illumio CloudSecure platform.",
+			Attributes: map[string]resource_schema.Attribute{
+				IDFieldName: idAttribute,
+				"application_id": resource_schema.StringAttribute{
+					Description: "ID of the application.",
+					Required:    true,
+				},
+				"account_id": resource_schema.StringAttribute{
+					Description: "ID of the AWS account.",
+					Required:    true,
+				},
+				"aws_arns": resource_schema.SetAttribute{
+					ElementType: types.StringType,
+					Description: "ARNs of AWS resources to associate with the Cloudsecure application",
+					Optional:    true,
+				},
+				"aws_directconnect_connections": resource_schema.SetAttribute{
```

Comment:
1. Use the same names as the corresponding 
1. Clarify in the name that this contains IDs.
1. Sets are a bit painful to Terraform users: it requires doing conversions to/from lists, etc. Using a `ListAttribute` would be more user-friendly. **Same for all the other `SetAttributes` in this resource.**
```suggestion
				"aws_dx_connection_ids": resource_schema.ListAttribute{
```

Reply:
But set is useful to check duplicates at the runtime.
Also https://developer.hashicorp.com/terraform/language/expressions/types#sets
the set is same as ["1","2"] when set on a attribute.
if a list is computed adding to_set should be simple.

please let me know if we should still favor list.

closed all other List based suggestions for single discussion thread.

Reply:
updated to list.